src/Course/List.hs

```diff
@@ -261,12 +251,9 @@ seqOptional =
 --
 -- >>> find (const True) infinity
 -- Full 0
-find ::
-  (a -> Bool)
-  -> List a
-  -> Optional a
-find =
-  error "todo: Course.List#find"
+find :: forall a as . (a -> Bool) -> List a -> Optional a
+find p as = headOr Empty (map (Full) (filter p as))
```

Comment:
Good job re-using filter and headOr!
```suggestion
find p as = headOr Empty (map Full (filter p as))
```

Reply:
Thanks, I was actually a little surprised it worked, I thought it might fail on the infinite example

Reply:
Everything is lazy, filter just filter as many elements as map requires, which in turn maps only as many as head requires, which is 1